src/v/config/configuration.cc

```diff
@@ -1662,7 +1662,34 @@ configuration::configuration()
       "balancer, in milliseconds",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       5000ms,
-      {.min = 1ms, .max = bottomless_token_bucket::max_width}) {}
+      {.min = 1ms, .max = bottomless_token_bucket::max_width})
+  , kafka_quota_balancer_node_period(
+      *this,
+      "kafka_quota_balancer_node_period_ms",
+      "Intra-node throughput quota balancer invocation period, in "
+      "milliseconds. Value of 0 disables the balancer and makes all the "
+      "thoughput quotas immutable.",
```

Comment:
Well those constants are per-shard ingress and egress quotas, currently they are based on `kafka_throughput_limit_node_in_bps` and `kafka_throughput_limit_node_out_bps` most of the time, although there is a corner case, when this property is changed on the fly, the value of the quotas are whatever the balancer has set before that. Moreover, once there is cluster throughput limit, those constants are also influenced by the cluster wide setting and the number of nodes in the cluster (in future, not in this iteration).

I'm not sure it makes sense to try squeezing that info into configuration property description -- please confirm that it does and I will change it.

Reply:
Thanks for clarifying. Before thinking about the wording, I'm interested in the functionality which I didn't realise existed.

Functionality i'm not sure it makes sense to leave the values fixed at whatever value they are at when the balancer is disabled. From a usability perspective I don't think I'd disable the balancer unless something bad was happening, and in that case, it might be because the balancer was misbehaving.

In this case I'd probably expect that when the balancer was disabled that the quotas reset to something sane.

wdyt?

Reply:
> In this case I'd probably expect that when the balancer was disabled that the quotas reset to something sane.

That makes sense. So do you suggest when we see the change of this property to `0`, we should reset quotas to the default values as well? That may make sense, however that may look like an unexpected side effect though. 

Currently the way to manually reset the quotas is to disable limiting by removing the `kafka_throughput_limit_node_*_bps` property and then by putting it back. It's not the most convenient way to do that.

I vote for that feature.
- do you agree?
- ok for a later PR?

Reply:
> disable limiting by removing the kafka_throughput_limit_node_*_bps property and then by putting it back

i'm not sure what that means. you'd have to change the config option and then change it again? that doesn't seem intuitive.

> ok for a later PR?

yes, please create a ticket to follow this up. it's important.

Reply:
https://github.com/redpanda-data/redpanda/issues/8392

Reply:
thank you!